cocotb/binary.py

```diff
@@ -303,12 +303,35 @@ def is_resolvable(self):
         """Does the value contain any ``X``'s?  Inquiring minds want to know."""
         return not any(char in self._str for char in BinaryValue._resolve_to_error)
 
-    value = property(get_value, set_value, None,
-                     "Integer access to the value. **deprecated**")
-    integer = property(get_value, set_value, None,
-                       "The integer representation of the underlying vector.")
-    signed_integer = property(get_value_signed, set_value, None,
-                              "The signed integer representation of the underlying vector.")
+    @property
+    def value(self):
+        "Integer access to the value. **deprecated**"
+        return self.get_value()
+
+    @value.setter
+    def value(self, val):
+        "Integer access to the value. **deprecated**"
+        self.set_value(val)
+
+    @property
+    def integer(self):
+        "The integer representation of the underlying vector."
+        return self.get_value()
+
+    @integer.setter
+    def integer(self, val):
+        "The integer representation of the underlying vector."
```

Comment:
You can leave off the docstrings on all of the setters, only the getter's docstring is used.